Update - August 2020 - New filters on product structure API endpoints (#739)
akeneo/pim-helpcenter: content/updates/2020-08/3-API-option-labels.md

---
pim_announcement_img: ./img/option-labels.png
pim_announcement_alt_img: "Option labels inside product endpoint"
pim_announcement_audience:
- CE
- EE
---

> # Get options labels directly inside the products endpoint

Reviewer: ```suggestion
# Get options labels directly inside the products endpoints
```
"products endpoints" instead of "products endpoint" ?

Reviewer: But only the Get product/products endpoints are concerned by this feature right?

Reviewer 2: I feel like talking about endpoints here won't necessarily be understandable for Julia. I would suggest something like:
```suggestion
# Options labels in product export via API
```

::: meta-data type="Improvement" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/api-reference.html#get_products"
:::

> To help you reduce the number of API calls needed to synchronize your product data, we also added option labels inside the product endpoint.

Reviewer 2: I would remove the "also" which sounds weird.

> To help you reduce the number of API calls needed to synchronize your product data, we also added option labels inside the product endpoint.

Author: I don't know for the product endpoints or products endpoints. 
Both are probably working.

> Let's take the PDF generation example. Before this update, the PDF generator needed to retrieve product attribute values by calling the `GET /products` endpoint. Then, for each attribute with a list of options, a `GET /attributes` was required to get the related labels. 

Reviewer 2: Good idea, Laure. Indeed, "the PDF generation example" can lead to a misunderstanding as it is also a native feature of the PIM.

> Let's take the PDF generation example. Before this update, the PDF generator needed to retrieve product attribute values by calling the `GET /products` endpoint. Then, for each attribute with a list of options, a `GET /attributes` was required to get the related labels. 

Author: Yes, good idea Laure!

---
pim_announcement_img: ./img/option-labels.png

> pim_announcement_alt_img: "Option labels inside product endpoint"

Reviewer: same applies here, are they needed?

pim_announcement_audience:
- CE
- EE
---

> # Get options labels directly inside the product endpoints

Reviewer 2: I feel like talking about endpoints here won't necessarily be understandable for Julia. I would suggest something like:
```suggestion
# Options labels in product export via API
```

::: meta-data type="Improvement" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/api-reference.html#get_products"
:::

> To help you reduce the number of API calls needed to synchronize your product data, we added option labels inside the product endpoints.

Author: I updated with `product endpoints` for both paragraphs.

Let's take an example! 

> If you use our API to display product content on a website, or to create marketing/sales materials, before this update, your connector needed to retrieve product attribute values by calling the `GET /products` endpoint. Then, for each attribute with a list of options, a `GET /attributes` was required to get the related labels. 

Author: @LaureBro to review with your idea.

> Before the generator was only receiving codes:

Reviewer 2: ```suggestion
Before the connector was only receiving codes:
```

                    "code": "s",
                    "labels": {
                        "en_US": "S",
                        "fr_FR": "S"
                    }
                }
            }
        ],
        "materials": [
            {
                "locale": null,
                "scope": null,
                "data": [
                    "leather",
                    "metal"
                ],
                "linked_data": {
                    "leather": {
                        "attribute": "materials",
                        "code": "leather",
                        "labels": {
                            "de_DE": "Leder",
                            "en_US": "Leather",
                            "fr_FR": "Cuir"
                        }
                    },
                    "metal": {
                        "attribute": "materials",
                        "code": "metal",
                        "labels": {
                            "de_DE": "Metall",
                            "en_US": "Metal",
                            "fr_FR": "Métal"
                        }
                    }
                }
            }
        ]
    }
```

> In that really simple example, we **avoid 5 API calls per product**. Imagine this improvement for a catalog which contains thousands of products! 

Reviewer: ```suggestion
In this example, we **avoid 5 API calls per product**. Imagine this improvement for a catalog which contains thousands of products! 
```

---
pim_announcement_img: ./img/option-labels.png
pim_announcement_alt_img: "Option labels inside product endpoint"
pim_announcement_audience:
- CE
- EE
---

# Get options labels directly inside the products endpoint
::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/api-reference.html#get_products"
:::

> We improved our API to help you reducing the number of API calls needed to synchronize product data.

Reviewer 2: ```suggestion
We improved our API to help you reduce the number of API calls needed to synchronize product data.  
```
Please, do not forget those two spaces at the end of the line.

Reviewer 2: Can we mention somewhere why we added specifically those labels? Because they usually need them to be displayed in their e-commerce front I guess?

Reviewer 2: Or the PDF example you gave in Slack could work too.

Author: I launched the help center on my computer and it's a bit weird with a line break.
I also added the PDF example that I gave you in Slack. 
But I don't really know how to format the attribute list with codes and labels. It will be super long if I use a line for each code then a new line per label. What do you think?

> We improved our API to help you reducing the number of API calls needed to synchronize product data.

Reviewer: ```suggestion
To help you reduce the number of API calls needed to synchronize your product data, we also added option labels inside the product endpoint.
```

> To do so, we added option labels inside the product endpoint.

Reviewer: ```suggestion
```
